apps/web/src/app/dashboard/[name]/settings/members/page.tsx

```diff
@@ -0,0 +1,309 @@
+
+  useEffect(() => {
+    refetchOrg();
+  }, [organization]);
+
+  const handleRoleChange = async (newValue: "user" | "owner" | "admin", userId: string) => {
+    if (newValue === "owner") {
+      setIsAlertOpen(true);
+      setNewRole({ memberId: userId, organizationId: organization!.id, role: newValue });
+      return;
+    }
+    try {
+      await updateMemberRole({ memberId: userId, organizationId: organization!.id, role: newValue });
+      refetchUsers();
+      toast({
+        title: "Success",
+        description: "User role was updated successfully",
+      });
+    } catch (error) {
+      if (error instanceof Error) {
+        toast({
+          title: "Error",
+          description: `${error.message} !`
+        });
+      }
+    }
+  }
+
+  const handleAlertAccepted = async () => {
+    setIsAlertOpen(false);
+    if (newRole) {
+      await updateMemberRole(newRole);
+      refetchUsers();
+      setNewRole(null);
+    }
+  }
+
+  const handleRemoveMember = async (userId: string) => {
+    try {
+      await removeMemberFromOrganization({ memberId: userId, organizationId: organization!.id });
+      toast({
+        title: "Success",
+        description: "User was removed successfully",
+      });
+    } catch (error) {
+      if (error instanceof Error) {
+        toast({
+          title: "Error",
+          description: `${error.message} !`
+        });
+      }
+    }
+  }
+
+  const handleLeave = async () => {
+    try {
+      await removeMemberFromOrganization({ memberId: session.data!.user.id, organizationId: organization!.id });
+      toast({
+        title: "Success",
+        description: "You have left the organization",
+      });
+      router.push("/dashboard");
+      refetchUsers();
+    }
+    catch (error) {
+      if (error instanceof Error) {
+        toast({
+          title: "Error",
+          description: `${error.message} !`
+        });
+      }
+    }
+  }
+
+
+  const form = useForm<Invite>({
+    resolver: zodResolver(inviteSchema),
+    defaultValues: {
+      email: "",
+    },
+  });
+
+  const onSubmit: SubmitHandler<Invite> = async (data) => {
+    await addMemberToOrganizationByUserEmail({
+      email: data.email,
+      organizationId: organization!.id,
+      role: "user",
+    })
+      .then(() => {
+        refetchUsers()
+        toast({
+          title: "Success",
+          description: `User ${data.email} was added successfully`,
+        });
+      })
+      .catch((error) => {
+        toast({
+          title: "Error",
+          description: ` ${error.message}!`,
+        });
+      })
+
+    setIsModalOpen(false);
+  };
+
+  const onError: SubmitErrorHandler<Invite> = (error) => {
+    console.log(error);
+    toast({
+      title: "Error",
+      description: `Failed to send and invitation: ${error.email!.message} !`,
+    });
+  };
+
+  return (
+    <>
+      {usersLoaded && (
```

Comment:
This fragment is unnscessary. Also add a loading state for this component. Like a spinner but preferably a skeleton component

Reply:
added skeletons to whole settings and nav. Idk if I like them tho.